store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java

```diff
@@ -70,95 +94,203 @@ public void setUp() throws Exception {
 
-        LIBRARY.addSchema(bSchema1ID, bSchema);
+        // Then
+        HashSet<String> expected = new HashSet<>();
+        expected.addAll(Arrays.asList("a", "b", "c"));
 
-        assertFalse(LIBRARY.exists("b"));
+        assertEquals(expected, graphIds);
+    }
 
-        fStore.execute(OperationChain.wrap(new AddGraph.Builder()
-                .graphId("b")
-                .parentPropertiesId(ACC_PROP)
-                .parentSchemaIds(Lists.newArrayList(bSchema1ID))
-                .build()), testContext);
+    @Test
+    public void shouldGetCorrectDefaultViewForAChosenGraphOperation() throws OperationException {
+        // Given
+        addGroupCollisionGraphs();
 
-        fStore.execute(OperationChain.wrap(new AddGraph.Builder()
-                .graphId("c")
-                .parentPropertiesId(ACC_PROP)
-                .parentSchemaIds(Lists.newArrayList(aSchema1ID))
-                .build()), testContext);
+        // When
+        final CloseableIterable<? extends Element> a = fStore.execute(new OperationChain.Builder()
+                .first(new GetAllElements.Builder()
+                        //No view so makes default view, should get only view compatible with graph "a"
+                        .option(FederatedStoreConstants.KEY_OPERATION_OPTIONS_GRAPH_IDS, "a")
+                        .build())
+                .build(), testContext);
 
-        // No exceptions thrown - as all 3 graphs should be able to be added together.
+        // Then
+        assertNotNull(a);
+        assertFalse(a.iterator().hasNext());
     }
 
     @Test
-    public void shouldGetCorrectDefaultViewForAChosenGraphOperation() throws
-            Exception {
-        final HashMapGraphLibrary library = new HashMapGraphLibrary();
-        library.add("a", new Schema.Builder()
-                .edge("e1", getProp("prop1"))
-                .type(DIRECTED_EITHER, Boolean.class)
-                .merge(STRING_SCHEMA)
-                .build(), PROPERTIES);
-
-        library.add("b", new Schema.Builder()
-                .edge("e1", getProp("prop2"))
-                .type(DIRECTED_EITHER, Boolean.class)
-                .merge(STRING_SCHEMA)
-                .build(), PROPERTIES);
+    public void shouldBeAbleToGetElementsWithOverlappingSchemas() throws OperationException {
+        // Given
+        addOverlappingPropertiesGraphs(false);
 
+        // Element 1
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
+                        .source("source1")
+                        .dest("dest1")
+                        .property("prop1", "value1")
+                        .build())
+                .build(), testContext);
 
-        fStore.execute(new AddGraph.Builder()
-                .graphId("a")
+        // Element 2
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
```

Comment:
This edge is used 3 times through the test class

Reply:
Decided I preferred it verbose